### DISCUSSION TITLE
Decrypting a stream in the browser eats the memory

### DISCUSSION THREAD

**Discussion by janosNH** [openpgpjs/openpgpjs#1317] [Q&A] [answered]
I'm trying to download a symmetrically encrypted large file from a webserver and decrypt it. In Node using streams for encryption and decryption works fine w/ any size of files. I use nearly identical code in the browser. It works w/ ~30MB files, but fails w/ ~700MB. I get this exception: 'Error decrypting message: Array buffer allocation failed'
The relevant part of my code:
```javascript
async function download () {  
    const fileStream = streamSaver.createWriteStream('your.img')  
    const res = await fetch(url)  
    const bodyStream = res.body
    const message = await openpgp.readMessage({binaryMessage: bodyStream})
    const decrypted = await openpgp.decrypt({
        message,
        passwords: pw,
        format: 'binary'
    })
    return decrypted.data.pipeTo(fileStream)
}
```
 I suspect readMessage tries to read everything from the stream. How can I avoid it?
Thanks for your help.

**Comment by twiss** [comment, marked as answer]
Hey :wave: Do you have `openpgp.config.allowUnauthenticatedStream` set? (To be fair, I'm not sure why this would only affect web streams, but just to check.) Failing that, the underlying issue you're running into may be the same as https://github.com/openpgpjs/openpgpjs/issues/1012. I'm not aware of a workaround, but that thread does contain some hints about where to start looking, if you want to dig into this.

**Reply by janosNH** [reply]
You saved my day! openpgp.config.allowUnauthenticatedStream = true works. In node I used file streams not fetch, maybe that's the difference.

Thanks for your quick help.

**Reply by twiss** [reply]
:+1: In that case, I suspect that Node.js was also allocating more memory than needed, but perhaps less noticeably so. But, I'm glad that that fixed it :)
